### PR TITLE
Add notification control for playback from foreground service.

### LINKED ISSUE #49: Add playback control from notification
When using a foreground service for file playback, it should create a notification to control the playback

### PR DESCRIPTION
**Pull request by flamme**
Fixes #49

### CODE REVIEW

File: app/src/main/java/com/google/android/soundchecker/FilePlayerService.kt [hunk 1]
```diff
@@ -19,31 +19,63 @@ package com.google.android.soundchecker
...
+            ACTION_START_SERVICE -> {
+                startForeground(
+                    NOTIFICATION_ID, getNotification(""),
+                    ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK
+                )
+            }
+            ACTION_PLAYBACK_STARTED -> {
+                mTitle = intent.getStringExtra(TITLE).toString()
+                mSongName = intent.getStringExtra(SONG_NAME).toString()
+                startForeground(NOTIFICATION_ID, getNotification(ACTION_STOP_PLAYBACK),
```
Comment (robertwu1) [inline app/src/main/java/com/google/android/soundchecker/FilePlayerService.kt:63]: Why is this one ACTION_STOP_PLAYBACK and the one below ACTION_START_PLAYBACK?

File: app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt [hunk 2]
```diff
@@ -339,4 +350,30 @@ open class BaseFilePlayerActivity : ComponentActivity(), OnAudioFocusChangeListe
...
+                }
+            }
+        }
+    }
+
+    private fun checkAndRequestNotificationPermission() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU &&
+            checkSelfPermission(POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
+                registerForActivityResult(
+                        ActivityResultContracts.RequestPermission()) { }.launch(POST_NOTIFICATIONS)
```
Comment (robertwu1) [inline app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt:376]: Do you get a callback for this? If so, can you make confirm that this code works both when the permission is already granted and when you need to wait for the notification?

File: app/src/main/java/com/google/android/soundchecker/FilePlayerService.kt (hunk 1 above)
Comment (flamme) [reply app/src/main/java/com/google/android/soundchecker/FilePlayerService.kt:63]: When the playback is started, the action from the notification is to stop it. When the playback is stopped, the action will now change to start the playback.

File: app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt (hunk 2 above)
Comment (flamme) [reply app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt:376]: Yes.

**Review by robertwu1 — APPROVED** [review]

File: app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt [hunk 3]
```diff
@@ -160,31 +164,28 @@ open class BaseFilePlayerActivity : ComponentActivity(), OnAudioFocusChangeListe
...
 
     override fun onDestroy() {
+        unregisterReceiver(mBecomingNoisyReceiver)
+        unregisterReceiver(mPlaybackStateChangedReceiver)
         mAudioManager.abandonAudioFocusRequest(mFocusRequest)
+        stopPlayback()
         mMsgHandlerThread.quitSafely()
+        val intent = Intent(this, FilePlayerService::class.java)
+        intent.action = FilePlayerService.ACTION_STOP_SERVICE
+        startForegroundService(intent)
```
Comment (philburk) [inline app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt:188]: This looks weird to me but I believe it is correct.

**Review by philburk — APPROVED** [review]

File: app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt (hunk 3 above)
Comment (flamme) [reply app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt:188]: Yes, that is correct. By calling `startForegoundService`, `onStartCommand` will be called. The intent is handled there. If the service is not running, `onCreate` for the service will be first called and then `onStartCommand` will be called after that.

**Pull request merged**